pkg/controllers/machine/termination/controller.go

```diff
@@ -86,7 +87,9 @@ func (c *Controller) Finalize(ctx context.Context, machine *v1alpha5.Machine) (r
 	if len(nodes) > 0 {
 		return reconcile.Result{}, nil
 	}
-	if machine.Status.ProviderID != "" {
+	// If the providerID is defined, use that value; else, use the machine linked annotation if it's on the machine
+	providerID := lo.Ternary(machine.Status.ProviderID != "", machine.Status.ProviderID, machine.Annotations[v1alpha5.MachineLinkedAnnotationKey])
```

Comment:
maybe a nitpick, so for your consideration only...

the vanilla CS 101 thing is a little more readable to me:

```
if machine.Status.ProviderID != ""
    || machine.Annotations[v1alpha5.MachineLinkedAnnotationKey] != "" {
...
```

If we were using the providerID string later in the flow then I think capturing it in that ternary fashion would make sense, but because we're not I sort of paused for a bit thinking there was more interesting going on then there is.

(Of course the fact that this is fixing an edge case is definitely interesting! :))